### PR TITLE
Update reference to macOS SDK extraction instructions

### PR DESCRIPTION
**Pull request by hebasto**
This PR replaces the outdated `gitian-building/gitian-building-mac-os-sdk.md` with the link to https://github.com/bitcoin/bitcoin/blob/master/contrib/macdeploy/README.md#sdk-extraction

### CODE REVIEW

**Comment by MarcoFalke** [comment]
ACK 1381c21745c00f614389ff75eb1fbe49b82ee53e

**Comment by MarcoFalke** [comment]
Should https://github.com/bitcoin/bitcoin/blob/master/contrib/macdeploy/extract-osx-sdk.sh be deleted as well?

**Comment by hebasto** [comment]
> Should https://github.com/bitcoin/bitcoin/blob/master/contrib/macdeploy/extract-osx-sdk.sh be deleted as well?

Correct. But it is in another repo :)
I was going to submit a pr after merging this one.

**Comment by MarcoFalke** [comment]
On a second thought, should the previous instructions (or a link to them) be preserved, because they are still needed for gitian builds of old tags?

**Comment by hebasto** [comment]
> On a second thought, should the previous instructions (or a link to them) be preserved, because they are still needed for gitian builds of old tags?

Is it enough for a reader to just switch the branch: for example https://github.com/bitcoin/bitcoin/blob/0.20/contrib/macdeploy/README.md ?

File: gitian-building.md [hunk 1]
```diff
@@ -76,6 +72,18 @@ git remote add $NAME git@github.com:$NAME/gitian.sigs.git
 
 Where `satoshi` is your GitHub name.
 
+macOS code signing
+------------------
+In order to sign builds for macOS, you need to download the free SDK and extract a file.
+The steps are described [here](https://github.com/bitcoin/bitcoin/blob/master/contrib/macdeploy/README.md#sdk-extraction).
```
Comment (MarcoFalke) [inline gitian-building.md:78]:
```suggestion
The steps are described [here](https://github.com/bitcoin/bitcoin/blob/master/contrib/macdeploy/README.md#sdk-extraction) for `Xcode-11.3.1-11C505-extracted-SDK-with-libcxx-headers.tar.gz`.
The steps are described [here](https://github.com/bitcoin-core/docs/blob/a706cc6afde0c8999fb0b8e3356522a7f8e2ad8f/gitian-building/gitian-building-mac-os-sdk.md) for `MacOSX10.11.sdk.tar.gz`.
```

**Review by MarcoFalke — COMMENTED** [review]
maybe like this?

File: gitian-building.md (hunk 1 above)
Comment (hebasto) [reply gitian-building.md]: Updated.

But a new list of the explicitly mentioned extracted SDKs misses `MacOSX10.14.sdk.tar.gz`.

**Comment by MarcoFalke** [comment]
ACK 0aebb726ba7abda71c796552a54401190e28a55e

**Pull request merged**